js/ui/discoveryFeedButton.js

```diff
@@ -54,21 +50,42 @@ function maybeCreateButton() {
  */
 const DiscoveryFeedButton = new Lang.Class({
     Name: 'DiscoveryFeedButton',
-    Extends: St.Button,
+    Extends: St.BoxLayout,
 
     _init: function() {
-        this.parent({ name: 'discovery-feed',
-                      child: new St.Icon({ icon_name: 'go-down-symbolic',
-                                           icon_size: 16,
-                                           style_class: 'discovery-feed-icon ' }),
-                      style_class: 'discovery-feed-button',
-                      visible: _primaryMonitorWidthPassesThreshold() });
+        this.parent({ vertical: true,
+                      style_class: 'discovery-feed-button' });
+
+        this._bar = new St.Button({ name: 'discovery-feed-bar',
+                                    child: new St.Icon({ style_class: 'discovery-feed-bar-icon' }),
+                                    style_class: 'discovery-feed-bar',
+                                    visible: _primaryMonitorWidthPassesThreshold() });
+        this.add(this._bar);
+
+        this._tile = new St.Button({ name: 'discovery-feed-tile',
+                                    child: new St.Icon({ style_class: 'discovery-feed-tile-icon' }),
+                                    style_class: 'discovery-feed-tile',
+                                    visible: _primaryMonitorWidthPassesThreshold() });
+        this.add(this._tile, { x_fill: false,
+                               x_align: St.Align.MIDDLE,
+                               expand: true });
+
+        this._bar.connect('event', Lang.bind(this, this._onEvent));
+        this._tile.connect('event', Lang.bind(this, this._onEvent));
```

Comment:
We don't appear to re-use `this._bar` and `this._tile` anywhere outside this function, so prefer to just make them variables inside this function.

Reply:
+1. Since the title and the bar are added as children of the box layout, there's no need to keep a reference to them

Reply:
Did we come to a resolution on this @erikos  ?

Reply:
Err, whoops, my bad, I see that we use them below.

Reply:
Yes, that has changed :)